### PR TITLE
Graduate Leader Migration to GA

### PR DESCRIPTION
**Pull request by jiahuif**
#### What type of PR is this?

/kind feature
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Leader Migration is now GA. All new configuration files onwards should use version v1.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by sftim** [comment]
I'd expect to see a changelog entry for this PR. Is that covered elsewhere?

**Comment by jiahuif** [comment]
Leader Migration requires an extra flag on each controller manager to activate. Flipping the feature gate will result in no difference of enabling or configuring this feature. I've added notes on new v1 types that was added alongside the feature gate.

/retest

**Comment by jpbetz** [comment]
/assign @cheftako @andrewsykim
for approval

**Comment by cheftako** [comment]
> I'd expect to see a changelog entry for this PR. Is that covered elsewhere?

https://github.com/kubernetes/kubernetes/pull/109072#issuecomment-1080893231 covers the config file change in a bit more detail. This change is only relevant to people running HA legacy cloud provider (Google, Amazon, Microsoft, VMWare) clusters who need to migrate from KCM to KCM + CCM. We've made the relevant cloud providers aware, as well as related parties (Redhat and KOPS). Happy for further suggestions on where to reach out.

**Comment by jpbetz** [comment]
Release leads- Okay if we keep this in 1.24? We have SIG approval we just need a /pkg/features approver, which was not realized yesterday?

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -969,7 +970,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 	KubeletInUserNamespace:                         {Default: false, PreRelease: featuregate.Alpha},
 	MemoryQoS:                                      {Default: false, PreRelease: featuregate.Alpha},
 	CPUManagerPolicyOptions:                        {Default: true, PreRelease: featuregate.Beta},
-	ControllerManagerLeaderMigration:               {Default: true, PreRelease: featuregate.Beta},
+	ControllerManagerLeaderMigration:               {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.26
```
Comment (andrewsykim) [inline pkg/features/kube_features.go:973]: Do we need to clean up this function and it's callers?

https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/staging/src/k8s.io/controller-manager/pkg/leadermigration/feature.go#L25-L28

**Review by andrewsykim — COMMENTED** [review]
Approving the `pkg/features` changes based on discussions in the SIG that this feature is ready for GA in v1.24

Deferring rest to @jpbetz and @cheftako.

/approve

File: pkg/features/kube_features.go (hunk 1 above)
Comment (jiahuif) [reply pkg/features/kube_features.go:973]: We can remove these while removing the feature gate itself. Otherwise it's just a no-op.

**Pull request merged**